react-components/src/apps/details/AppInfoDialog.js

```diff
@@ -115,18 +136,52 @@ function AppInfoDialog(props) {
         presenter.onClose();
     };
 
+    const onCreateQuickLaunch = () => {
+        presenter.onRequestToCreateQuickLaunch(app.id);
+        onClose();
+    };
+
+    const useQuickLaunch = (qLaunch) => {
+        presenter.getAppInfoForQuickLaunch(qLaunch.id, qLaunch.app_id);
+        onClose();
+    };
+
+    const doDeleteQuickLaunch = (qLaunch) => {
+        setLoading(true);
+        presenter.deleteQuickLaunch(
+            qLaunch.id,
+            () => {
+                let pos = quickLaunches.indexOf(qLaunch);
+                if (pos !== -1) {
```

Comment:
I wrote the check inside the callback because `doDeleteQuickLaunch` cannot be invoked without a quick launch in the first place. So i am making sure the quick launch is still there after the remote call is executed.

Reply:
> I had a few minor comments that aren't necessary for this PR, but I don't think I saw many static IDs for testing. Also, I think a few classes were missing prop-types. Other than that, LGTM 👍

I totally forgot about `ids`. Will add them 👍